securemailbox/views/retrieve.py

```diff
@@ -1,11 +1,77 @@
-from flask import Blueprint, jsonify
+from flask import Blueprint, jsonify, request, json
+from sqlalchemy.exc import IntegrityError
+
+from ..models import Mailbox
+from ..models import Message
 
 from securemailbox import db
 
 # Create the `retrieve` blueprint
 retrieve_blueprint = Blueprint("retrieve", __name__)
 
-
 @retrieve_blueprint.route("/retrieve/", methods=["POST"])
 def retrieve():
-    return jsonify({"success": True, "error": None})
+
+    # Check for valid json request
+    if not request.is_json:
+        return jsonify({'success': False, 'error': 'Request must be valid json', 'data': None}), 400
+
+    req_fingerprint = request.json.get('fingerprint', None)
+
+    # Check if fingerprint is present in json request body
+    if req_fingerprint is None:
+        return jsonify({'success': False, 'error': 'fingerprint not in request', 'data': None}), 400
+
+    # Check if fingerprint value is string variable
+    if not isinstance(req_fingerprint, str):
+        return jsonify({'success': False, 'error': 'fingerprint value not valid', 'data': None}), 400
+
+    # search for fingerprint in mailbox table
+    try:
+        mailbox_id = Mailbox.query.filter_by(fingerprint=req_fingerprint).first()
+    except IntegrityError:
```

Comment:
You probably want to catch a [`NoResultFound`](https://docs.sqlalchemy.org/en/13/orm/exceptions.html#sqlalchemy.orm.exc.NoResultFound) error here. Although I'm fairly certain by default `.first()` will return None instead of throwing an exception.